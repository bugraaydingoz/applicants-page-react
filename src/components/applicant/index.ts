export { Applicant } from './Applicant'
export { ApplicantList } from './ApplicantList'
