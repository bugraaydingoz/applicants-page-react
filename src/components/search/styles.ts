import styled from 'styled-components'

export const SearchContainer = styled.div`
  display: flex;
  justify-content: space-between;
  padding: 10px;
  width: 100%;
  height: 45px;
  border: 1px solid var(--gray-300);
  border-radius: var(--border-radius-lg);

  @media (min-width: 768px) {
    width: 305px;
    margin-right: var(--space-md);
  }
`

export const Input = styled.input`
  padding: 0 10px;
  width: 100%;
  font-size: 16px;
  border: none;
  color: var(--gray-400);
`
