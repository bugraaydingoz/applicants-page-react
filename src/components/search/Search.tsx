import React, { useState, useEffect } from 'react'

// Components
import { ReactComponent as SearchIcon } from '../shared/icons/search.svg'

// Hooks
import { useUrlQuery } from '../../hooks/useUrlQuery'

// Styles
import { SearchContainer, Input } from './styles'

export function Search() {
  const [urlQuery, setUrlQuery] = useUrlQuery()
  const [search, setSearch] = useState(urlQuery.search ?? '')

  useEffect(() => {
    if (!urlQuery.search?.length) {
      return
    }
    setSearch(urlQuery.search)
  }, [urlQuery.search])

  return (
    <SearchContainer>
      <SearchIcon />
      <Input
        type="text"
        placeholder="Search for applicant"
        value={search}
        onChange={(e) => setSearch(e.target.value)}
        onKeyPress={(e) => e.key === 'Enter' && setUrlQuery({ search })}
      />
    </SearchContainer>
  )
}
