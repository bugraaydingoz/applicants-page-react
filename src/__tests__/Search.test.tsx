import React from 'react'
import renderer from 'react-test-renderer'
import { Search } from '../components/search'

jest.mock('../hooks/useUrlQuery', () => ({
  useUrlQuery: jest.fn().mockImplementation(() => [{}, jest.fn()]),
}))

test('renders correctly', () => {
  const tree = renderer.create(<Search />).toJSON()
  expect(tree).toMatchSnapshot()
})
