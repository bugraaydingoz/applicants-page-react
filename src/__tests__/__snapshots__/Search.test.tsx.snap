// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-AxjAm hAIJkN"
>
  <svg>
    search.svg
  </svg>
  <input
    className="sc-AxirZ klVCza"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Search for applicant"
    type="text"
    value=""
  />
</div>
`;
